drivers/display/ssd_1306/ssd_1306.c

```diff
@@ -0,0 +1,371 @@
+			 0x0036494949360000,
+			 0x00060949291e0000,
+			 0x0000001200000000,
+			 0x0000005230000000,
+			 0x0000081414220000,
+			 0x0014141414141400,
+			 0x0000221414080000,
+			 0x0002015905020000,
+			 0x3e415d554d512e00,
+			 0x407c4a094a7c4000,//A
+			 0x417f494949493600,//B
+			 0x1c22414141412200,//C
+			 0x417f414141221c00,
+			 0x417f49495d416300,
+			 0x417f49091d010300,
+			 0x1c224149493a0800,
+			 0x417f0808087f4100,
+			 0x0041417F41410000,
+			 0x304041413F010100,
+			 0x417f080c12614100,
+			 0x417f414040406000,
+			 0x417f420c427f4100,
+			 0x417f420c117f0100,
+			 0x1c22414141221c00,
+			 0x417f490909090600,
+			 0x0c12212161524c00,
+			 0x417f090919694600,
+			 0x6649494949493300,
+			 0x0301417f41010300,
+			 0x013f4140413f0100,
+			 0x010f3140310f0100,
+			 0x011f6114611f0100,
+			 0x4141360836414100,
+			 0x0103447844030100,
+			 0x4361514945436100,
+			 0x00007f4141000000,
+			 0x0102040810204000,
+			 0x000041417f000000,
+			 0x0004020101020400,
+			 0x0040404040404000,
+			 0x0001020000000000,
+			 0x00344a4a4a3c4000,
+			 0x00413f4848483000,
+			 0x003c424242240000,
+			 0x00304848493f4000,
+			 0x003c4a4a4a2c0000,
+			 0x0000487e49090000,
+			 0x00264949493f0100,
+			 0x417f480444784000,
+			 0x0000447d40000000,
+			 0x000040443d000000,
+			 0x417f101824424200,
+			 0x0040417f40400000,
+			 0x427e027c027e4000,
+			 0x427e4402427c4000,
+			 0x003c4242423c0000,
+			 0x00417f4909090600,
+			 0x00060909497f4100,
+			 0x00427e4402020400,
+			 0x00644a4a4a360000,
+			 0x00043f4444200000,
+			 0x00023e4040227e40,
+			 0x020e3240320e0200,
+			 0x021e6218621e0200,
+			 0x4262140814624200,
+			 0x0143453805030100,
+			 0x004662524a466200,
+			 0x0000083641000000,
+			 0x0000007f00000000,
+			 0x0000004136080000,
+			 0x0018080810101800,
+			 0xAA55AA55AA55AA55
+			};
+
+/******************************************************************************/
+/************************** Functions Implementation **************************/
+/******************************************************************************/
+
+/***************************************************************************//**
+ * @brief Initializes ssd_1306 for display screening.
+ *
+ * @param device - The device structure.
+ * @return Returns SUCCESS in case of success or negative error code otherwise.
+*******************************************************************************/
+int32_t ssd_1306_init(struct display_dev *device)
+{
+	int32_t	ret;
+	uint8_t command[3];
+	ssd_1306_extra *extra;
+
+	extra = device->extra;
+	ret = gpio_get(&extra->dc_pin, extra->dc_pin_ip);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_get(&extra->reset_pin, extra->reset_pin_ip);
+	if (ret != SUCCESS)
+		return FAILURE;
+
+	// initial pin state
+	ret = gpio_direction_output(extra->dc_pin, DC_CMD);
+	if (ret != SUCCESS)
+		return FAILURE;
+	ret = gpio_direction_output(extra->reset_pin, RST_OFF);
+	if (ret != SUCCESS)
+		return FAILURE;
+	usleep(2000U);
```

Comment:
as a general rule, always document these delays by adding a comment of why is it needed, something like:
`Post reset delay, Treset=2ms`
where Treset is documented in datasheet